service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/auth/TokenCredentialCache.java

```diff
@@ -6,33 +6,57 @@
 import com.azure.core.credential.AccessToken;
 import com.azure.core.credential.TokenCredential;
 import com.azure.core.credential.TokenRequestContext;
+import lombok.extern.slf4j.Slf4j;
 
 import java.time.Duration;
 import java.time.Instant;
 import java.util.Objects;
 
 /**
- * This class generates AAD authentication tokens from a TokenCredential but caches previous tokens when they aren't near
- * expiry.
+ * This class generates AAD authentication tokens from a TokenCredential but caches previous tokens when they aren't
+ * near expiry.
  */
+@Slf4j
 public class TokenCredentialCache
 {
     private final static int MINUTES_BEFORE_PROACTIVE_RENEWAL = 9;
     private final TokenCredential tokenCredential;
     private AccessToken accessToken;
+    private final String[] authenticationScopes;
 
-    public static final String[] IOTHUB_PUBLIC_SCOPE = new String[]{"https://iothubs.azure.net/.default"};
-    public static final String BEARER_TOKEN_PREFIX = "Bearer ";
+    private static final String[] IOTHUB_PUBLIC_SCOPES = new String[]{"https://iothubs.azure.net/.default"};
+    private static final String[] IOTHUB_FAIRFAX_SCOPES = new String[]{"https://iothubs.azure.us"};
```

Comment:
How might a customer specify a value, if they are using yet another private cloud (black forest, mooncake, etc.)?

Reply:
My bad. Your comments specify that other private clouds don't require their own.

Reply:
They don't have to. The service team has confirmed for us that Fairfax is the only special case here.